cluster-autoscaler/estimator/threshold_based_limiter.go

```diff
@@ -56,9 +60,24 @@ func (tbel *thresholdBasedEstimationLimiter) PermissionToAddNode() bool {
 // after either a node count- of time-based threshold is reached. This is meant to prevent cases
 // where binpacking of hundreds or thousands of nodes takes extremely long time rendering CA
 // incredibly slow or even completely crashing it.
-func NewThresholdBasedEstimationLimiter(maxNodes int, maxDuration time.Duration) EstimationLimiter {
+func NewThresholdBasedEstimationLimiter(limits []BinpackingLimit, maxDuration time.Duration) EstimationLimiter {
+	maxNodes := computeLimit(limits, 0)
 	return &thresholdBasedEstimationLimiter{
-		maxNodes:    maxNodes,
-		maxDuration: maxDuration,
+		maxNodes:        maxNodes,
+		maxNodesDefault: maxNodes,
+		maxDuration:     maxDuration,
+	}
+}
+
+func computeLimit(binpackingLimits []BinpackingLimit, currentLimit int) int {
```

Comment:
This is ambiguous, should be change to `getNodeLimit`.

Reply:
Renamed in 7654f75852e2f3bbc63be0a199f53faa38bc3895